src/metrics/ranking.rs

```diff
@@ -9,39 +9,30 @@ use array::prelude::*;
 /// Return (nondecreasing) counts of true positives and false positives.
 fn counts_at_score(y_true: &[f32], y_hat: &[f32]) -> (Vec<f32>, Vec<f32>) {
 
-    // Sort scores and labels in a descending manner
-    let mut pairs = Vec::with_capacity(y_true.len());
-
-    for (&yt, &yh) in y_true.iter().zip(y_hat.iter()) {
-        pairs.push((yh, yt));
-    }
+    // vector of pairs (score, label) - the order is switched with respect to function arguments
+    let mut pairs: Vec<_> = y_hat.iter().cloned().zip(y_true.iter().cloned()).collect();
 
     pairs.sort_by(|a, b| b.0.partial_cmp(&a.0).unwrap_or(Ordering::Equal));
 
-    // Move down the scores; scores greater than the threshold are positives.
-    let mut true_positives = Vec::with_capacity(y_true.len());
-    let mut false_positives = Vec::with_capacity(y_true.len());
-    let mut positive_counter = 0.0;
-
-    let mut scores = Vec::new();
-
-    let mut prev_score = f32::INFINITY;
-
-    for (i, &(score, label)) in pairs.iter().enumerate() {
-
-        positive_counter += label;
-
-        if close(score, prev_score) {
-            continue;
+    // tp .. true positives, fp .. false positives
+    let mut s0 = ::std::f32::NAN;
+    let (mut tp, mut fp) = (0.0f32, 0.0f32);
+    let (mut tps, mut fps) = (vec![], vec![]);
+    for &(s, t) in pairs.iter() {
```

Comment:
Could you use slightly more descriptive names for the `s` and `t` variables? I appreciate that this is the convention in languages like Go and C, but I'm not greatly fond of it.